lfs/ntlm.go

```
@@ -0,0 +1,172 @@
+package lfs
+
+import (
+	"bytes"
+	"encoding/base64"
+	"github.com/ThomsonReutersEikon/go-ntlm/ntlm"	
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+func (c *Configuration) NTLMSession(creds Creds) ntlm.ClientSession {
+	
+	if c.ntlmSession != nil {
+		return c.ntlmSession
+	}
+	
+	splits := strings.Split(creds["username"], "\\")
+	var session, _  = ntlm.CreateClientSession(ntlm.Version2, ntlm.ConnectionOrientedMode)
+	session.SetUserInfo(splits[1], creds["password"], strings.ToUpper(splits[0]))
+	
+	c.ntlmSession = session
+	
+	return session
+}
+
+func DoNTLMRequest(request *http.Request, retry bool) (*http.Response, error) {					
+	handReq := cloneRequest(request)	
+	res, nil := InitHandShake(handReq)
```

I think this is silently discarding a possible error too.